### ISSUE TITLE
The sourcelist option cannot be specified for stage-in

### ISSUE THREAD

**Issue by rkmrHonjo** [DDNStorage/slurm#8]
# CommitID

- 0a5419d

# Issue

I specified the sourcelist option for stage-in.
But, it was ignored.

I used the following job script.

- jobscript
    ```
    #!/bin/bash

    #SBATCH --job-name=test
    #SBATCH --partition=debug
    #SBATCH --output=test-list-%j.out
    #SBATCH --nodelist=lod-node1,lod-node2
    #LOD setup
    #LOD stage_in --sourcelist=/mnt/lustre/test_list/stageinlist --source=/mnt/lustre/test_list/ --destination=/lod/
    #LOD stage_out --sourcelist=/mnt/lustre/test_list/stageoutlist --source=/lod/ --destination=/mnt/lustre/test_list/stage_out
    #LOD stop

    echo "test_list" 
    echo "stage out!!" > /lod/stage_out.txt
    ls -al /lod/
    ```

In slurmctld.log, the sourcelist option was not specified for stage-in.

- slurmctld.log
    ```
    [2019-08-02T06:53:42.365] bb_p_job_validate: burst_buffer:
    #LOD setup
    #LOD stage_in --sourcelist=/mnt/lustre/test_list/stageinlist --source=/mnt/lustre/test_list/ --destination=/lod/
    #LOD stage_out --sourcelist=/mnt/lustre/test_list/stageoutlist --source=/lod/ --destination=/mnt/lustre/test_list/stage_out
    #LOD stop
    [2019-08-02T06:53:42.365] debug2: LOD_DEBUG : in bb_p_job_validate after parsing
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG : in bb_p_job_validate2
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts in stage_in
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sin_src=/mnt/lustre/test_list/

    *** _parse_bb_opts could recognize the sourcelit option. ***
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sin_srclist=/mnt/lustre/test_list/stageinlist
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sin_dest=/lod/
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts in stage_out
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sout_src=/lod/

    *** I think that sout_srclist is correct. ***
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sin_srclist=/mnt/lustre/test_list/stageoutlist
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found sout_dest=/mnt/lustre/test_list/stage_out
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts in stage_out
    [2019-08-02T06:53:42.366] debug2: LOD_DEBUG: _parse_bb_opts return
    [･･･]
    *** The sourcelist option is not specified for stage-in. ***
    [2019-08-02T06:54:59.045] debug2: LOD_DEBUG: command: 
    [2019-08-02T06:54:59.045] debug2: lod
    [2019-08-02T06:54:59.045] debug2: --node=lod-node[1-2]
    [2019-08-02T06:54:59.045] debug2: --source=/mnt/lustre/test_list/
    [2019-08-02T06:54:59.045] debug2: --destination=/lod/
    [2019-08-02T06:54:59.045] debug2: stage_in
    [･･･]
    [2019/08/02-06:55:01] [root] [DEBUG] [__init__.py:1619] run command [mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve /mnt/lustre/test_list/ /lod/]
    [2019/08/02-06:55:01] [root] [DEBUG] [__init__.py:1619] starting [mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve /mnt/lustre/test_list/ /lod/] on host [lod-node1]
    [2019/08/02-06:55:04] [root] [DEBUG] [__init__.py:1619] command [ssh -a -x -l root -o StrictHostKeyChecking=no -o BatchMode=yes -q  lod-node1 "LANG=en_US.UTF-8 mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve /mnt/lustre/test_list/ /lod/"] finished, ret = [0], stdout = [[2019-08-02T06:55:00] Using sparse file
    [･･･]
    [2019-08-02T06:55:04.681] debug2: LOD_DEBUG: command:
    [2019-08-02T06:55:04.681] debug2: lod
    [2019-08-02T06:55:04.681] debug2: --node=lod-node[1-2]

    *** The sourcelist option is specified for stage-out. ***
    [2019-08-02T06:55:04.681] debug2: --sourcelist=/mnt/lustre/test_list/stageoutlist
    [2019-08-02T06:55:04.681] debug2: --source=/lod/
    [2019-08-02T06:55:04.681] debug2: --destination=/mnt/lustre/test_list/stage_out
    [2019-08-02T06:55:04.681] debug2: stage_out
    (･･･)
    [2019/08/02-06:55:06] [root] [DEBUG] [__init__.py:1619] run command [mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve --input /mnt/lustre/test_list/stageoutlist /lod/ /mnt/lustre/test_list/stage_out]
    [2019/08/02-06:55:06] [root] [DEBUG] [__init__.py:1619] starting [mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve --input /mnt/lustre/test_list/stageoutlist /lod/ /mnt/lustre/test_list/stage_out] on host [lod-node1]
    [2019/08/02-06:55:08] [root] [DEBUG] [__init__.py:1619] command [ssh -a -x -l root -o StrictHostKeyChecking=no -o BatchMode=yes -q  lod-node1 "LANG=en_US.UTF-8 mpirun -np 4  --host lod-node1,lod-node2 dcp --sparse --preserve --input /mnt/lustre/test_list/stageoutlist /lod/ /mnt/lustre/test_list/stage_out"] finished, ret = [0], stdout = [[2019-08-02T06:55:05] Using sparse file
    ```

**Comment by KnightKu** [comment]
@rkmrHonjo  Thanks for reporting this, let me have a look first.

**Comment by KnightKu** [comment]
Oh, the problem is sourcelist option is missing when launching stage-in command, it's a mistake here, fixed under commit aa4e0a7, please checkout and try it again, thanks.

**Comment by rkmrHonjo** [comment]
@KnightKu 
Thank you for rapid fixing.
I retry with it.

**Comment by rkmrHonjo** [comment]
@KnightKu 
Your fix works correctly. Thanks.

But, the following marked log message has not been fixed.

I think that the correct option is "sout_srclist" instead of "sin_srclist".

Please confirm it.

```
[2019-08-14T02:18:28.419] bb_p_job_validate: burst_buffer:
#LOD setup
#LOD stage_in --sourcelist=/mnt/lustre/test_list/stageinlist --source=/mnt/lustre/test_list/ --destination=/lod/
#LOD stage_out --sourcelist=/mnt/lustre/test_list/stageoutlist --source=/lod/ --destination=/mnt/lustre/test_list/stage_out/
#LOD stop
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG : in bb_p_job_validate after parsing
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG : in bb_p_job_validate2
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts in stage_in
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found sin_src=/mnt/lustre/test_list/
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found sin_srclist=/mnt/lustre/test_list/stageinlist
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found sin_dest=/lod/
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts in stage_out
[2019-08-14T02:18:28.419] debug2: LOD_DEBUG: _parse_bb_opts found sout_src=/lod/
****************************************************************************************
[2019-08-14T02:18:28.420] debug2: LOD_DEBUG: _parse_bb_opts found sin_srclist=/mnt/lustre/test_list/stageoutlist 
****************************************************************************************
[2019-08-14T02:18:28.420] debug2: LOD_DEBUG: _parse_bb_opts found sout_dest=/mnt/lustre/test_list/stage_out/
[2019-08-14T02:18:28.420] debug2: LOD_DEBUG: _parse_bb_opts found Lustre On Demand
[2019-08-14T02:18:28.420] debug2: LOD_DEBUG: _parse_bb_opts in stage_out
[2019-08-14T02:18:28.420] debug2: LOD_DEBUG: _parse_bb_opts return
[...]

**Comment by KnightKu** [comment]
Indeed, better to name flist option to sout_srclist for stage_out, otherwise may cause confusion to user.
Thanks for your suggestion, let me fix it now.

**Comment by KnightKu** [comment]
Fixed the typo in commit f70d03a.

**Comment by rkmrHonjo** [comment]
Thank you!
I close this issue.
